sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResource.cs

```diff
@@ -41,7 +40,13 @@ internal class AppendBlobStorageResource : StorageResourceItemInternal
         ///
         /// Will return default if the length was not set by a GetStorageResources API call.
         /// </summary>
-        protected override long? Length => _length;
+        protected override long? Length => ResourceProperties?.ContentLength;
+
+        /// <summary>
+        /// ETag Download lock. Used when reading from the storage resource to confirm the storage resource has not changed
+        /// since the start of the transfer.
+        /// </summary>
+        protected ETag? ETagDownloadLock;
```

Comment:
Hm is it okay even though StorageResource also encompasses `Local` Storage Resources that don't have an ETag?

Or should the exception just be as long as it's a HTTP concept and not an Azure specific concept we should be fine.

Reply:
Hmm, I didn't think about local resources. IMO, I think since its an HTTP concept, it should be fine. The other thing we could do is doing something more generic like "Version" or "VersionTag" or something and then just having it be a string. I still don't think we would use that for local but it would be a more generic concept that we could somehow use in Local if we needed.

That may be overkill though, I'm fine with Etag. I guess unless the `Etag` type is from Blob? Then maybe we would want a string or something. If its from Core, then its okay.

Reply:
done

Reply:
I still don't think you should save the Etag to the class on construction as the ResourceProperties will not always be provided on construction (like for single transfer (or currently rehydrate)). You should use `ResourceProperties.Etag` wherever Etag is used instead so if it is populated later by `GetProperties` we will get it.

Reply:
Just to be clear, we're going to save the ETag on the `UriToStreamPart` after we do a `GetProperties` call right?

Reply:
Using ETag from the StorageResourceItem.ResourceProperties